src/ResourceManager/Intune/Commands.Intune/Apps/AddIntuneAndroidMAMPolicyAppCmdlet.cs

```diff
@@ -0,0 +1,75 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace Microsoft.Azure.Commands.Intune
+{
+    using Management.Intune;
+    using Management.Intune.Models;
+    using Microsoft.Azure.Commands.Intune.Properties;
+    using System.Globalization;
+    using System.Management.Automation;
+
+    /// <summary>
+    /// A cmdlet to link an app to Android Intune MAM policy Azure resource.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Add, "AzureRmIntuneAndroidMAMPolicyApp", SupportsShouldProcess = true), OutputType(typeof(PSObject))]
+    public sealed class AddIntuneAndroidMAMPolicyAppCmdlet : IntuneBaseCmdlet
+    {
+        /// <summary>
+        /// Gets or sets the policy name
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The Android policy name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the App name
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The Android App name to link to.")]
+        [ValidateNotNullOrEmpty]
+        public string AppName { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "Don't ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+        /// <summary>
+        /// Executes the cmdlet.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            this.ConfirmAction(
+                this.Force,
+                string.Format(
+                    CultureInfo.CurrentCulture, 
+                    Resources.AddLinkedResouce_ActionMessage, 
+                    Resources.App, 
+                    this.AppName, 
+                    Resources.AndroidPolicy, 
+                    this.Name),
+                string.Format(
+                    CultureInfo.CurrentCulture, 
+                    Resources.AddLinkedResources_ProcessMessage, 
+                    Resources.App, 
+                    this.AppName, 
+                    Resources.AndroidPolicy, 
+                    this.Name),
+                this.Name,
+                () =>
+                {
+                    var payLoad = AppOrGroupPayloadMaker.PrepareMAMPolicyPayload(this.IntuneClient, LinkType.App, this.AsuHostName, this.AppName);
+                    this.IntuneClient.Android.AddAppForMAMPolicy(this.AsuHostName, this.Name, this.AppName, payLoad);
+                    this.WriteObject(Resources.OperationCompletedMessage);
```

Comment:
You should not write messages to the output stream.  Write messages to verbose or progress streams.  If you are returningnothing, then do not write anything.  Think of the output stream as the return value of the cmdlet.